### PR TITLE
Audited the javadoc in dynamodb-enhanced and corrected some error.

### PR DESCRIPTION
**Pull request by Quanzzzz**
Audit the javadoc in dynamodb-enhanced, corrected some bad formatted examples and some javadoc errors.
## Testing

Tested all the Javadocs by generator in IntelliJ.
## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/ListAttributeConverter.java [hunk 1]
```diff
@@ -45,28 +45,36 @@
...
+ * {@literal AttributeConverter<Set<Integer>> setConverter =
+ * CollectionAttributeConverter.setConverter(IntegerAttributeConverter.create());}
+ * </code>
  * <p>
- * AttributeConverter<SortedSet<Integer>> sortedSetConverter =
- * CollectionAttributeConverter.sortedSetConverter(IntegerAttributeConverter.create());
+ * <code>
+ * {@literal AttributeConverter<SortedSet<Integer>> sortedSetConverter =
+ * CollectionAttributeConverter.sortedSetConverter(IntegerAttributeConverter.create());}
  * </code>
```
Comment (millems) [inline services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/ListAttributeConverter.java:78]: What's the difference between <code> and <pre>?

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/MapAttributeConverter.java [hunk 2]
```diff
@@ -47,25 +47,28 @@
...
+ * <p>
  * <code>
- * AttributeConverter<Map<MonthDay, String>> mapConverter =
+ * {@literal AttributeConverter<Map<MonthDay, String>> mapConverter =
  * MapAttributeConverter.builder(EnhancedType.mapOf(Integer.class, String.class))
  * .mapConstructor(HashMap::new)
  * .keyConverter(MonthDayStringConverter.create())
  * .valueConverter(StringAttributeConverter.create())
- * .build();
+ * .build();}
```
Comment (millems) [inline services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/MapAttributeConverter.java:57]: Why @literal and not @code?

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/annotations/DynamoDbBean.java [hunk 3]
```diff
@@ -46,9 +46,11 @@
  * </ul>
  *
  * Example using attribute converter providers with one custom provider and the default provider:
+ * <pre>
  * {@code
  * (converterProviders = {CustomAttributeConverter.class, DefaultAttributeConverterProvider.class});
  * }
```
Comment (millems) [inline services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/annotations/DynamoDbBean.java:52]: Does @code work when the content has a } in it?

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/ListAttributeConverter.java (hunk 1 above)
Comment (Quanzzzz) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/ListAttributeConverter.java:78]: ```<code>``` makes the font different, but it cannot guarantee the formats, and ```<pre>``` can let the format of the code be showed as what they are in the original code file, i.e showing the indents and spaces.

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/MapAttributeConverter.java (hunk 2 above)
Comment (Quanzzzz) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/MapAttributeConverter.java:57]: The ```@literal``` is for the ```'<'``` and``` '>'``` signs, they are special signs in html, thus they cannot be recognize as the other signs, adding ```@literal``` guarantees that all the strings contained are just what they should be looked like. Usually this annotation is for special signs i.e ```'<', '>', '@'```

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/annotations/DynamoDbBean.java (hunk 3 above)
Comment (Quanzzzz) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/annotations/DynamoDbBean.java:52]: Only if all the brackets are matched, if not, there will be an error.

**Pull request merged**